src/command/backup/backup.c

```diff
@@ -0,0 +1,1904 @@
+    FUNCTION_TEST_END();
+
+    // Move (forward or back) to the next queue
+    queueIdx += clientIdx % 2 ? -1 : 1;
+
+    // Deal with wrapping on either end
+    if (queueIdx < 0)
+        FUNCTION_TEST_RETURN((int)queueTotal - 1);
+    else if (queueIdx == (int)queueTotal)
+        FUNCTION_TEST_RETURN(0);
+
+    FUNCTION_TEST_RETURN(queueIdx);
+}
+
+// Callback to fetch backup jobs for the parallel executor
+typedef struct BackupJobData
+{
+    const String *const backupLabel;                                // Backup label (defines the backup path)
+    const bool backupStandby;                                       // Backup from standby
+    const String *const cipherSubPass;                              // Passphrase used to encrypt files in the backup
+    const bool compress;                                            // Is the backup compressed?
+    const unsigned int compressLevel;                               // Compress level if backup is compressed
+    const bool delta;                                               // Is this a checksum delta backup?
+    const uint64_t lsnStart;                                        // Starting lsn for the backup
+
+    List *queueList;                                                // List of processing queues
+} BackupJobData;
+
+static ProtocolParallelJob *backupJobCallback(void *data, unsigned int clientIdx)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM_P(VOID, data);
+        FUNCTION_TEST_PARAM(UINT, clientIdx);
+    FUNCTION_TEST_END();
+
+    ASSERT(data != NULL);
+
+    ProtocolParallelJob *result = NULL;
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        // Get a new job if there are any left
+        BackupJobData *jobData = data;
+
+        // Determine where to begin scanning the queue (we'll stop when we get back here).  When copying from the primary during
+        // backup from standby only queue 0 will be used.
+        unsigned int queueOffset = jobData->backupStandby && clientIdx > 0 ? 1 : 0;
+        int queueIdx = jobData->backupStandby && clientIdx == 0 ?
+            0 : (int)(clientIdx % (lstSize(jobData->queueList) - queueOffset));
+        int queueEnd = queueIdx;
+
+        do
+        {
+            List *queue = *(List **)lstGet(jobData->queueList, (unsigned int)queueIdx + queueOffset);
+
+            if (lstSize(queue) > 0)
+            {
+                const ManifestFile *file = *(ManifestFile **)lstGet(queue, 0);
+
+                // Create backup job
+                ProtocolCommand *command = protocolCommandNew(PROTOCOL_COMMAND_BACKUP_FILE_STR);
+
+                protocolCommandParamAdd(command, VARSTR(manifestPathPg(file->name)));
+                protocolCommandParamAdd(
+                    command, VARBOOL(!strEq(file->name, STRDEF(MANIFEST_TARGET_PGDATA "/" PG_PATH_GLOBAL "/" PG_FILE_PGCONTROL))));
+                protocolCommandParamAdd(command, VARUINT64(file->size));
+                protocolCommandParamAdd(command, file
```

Comment:
(Line 1795?) The sync backup path at the end of the backup in Perl used to sync the backup LABEL path - which in C it appears we are syncing everything in the backup repo? There were also some conditionals in Perl (e.g. backup=full or pathExists) which I don't see here.  Also the sync was performed before we saved the manifest (which did not seem like that was right anyway).
```
    # Sync backup path if supported
    if ($oStorageRepo->capability(STORAGE_CAPABILITY_PATH_SYNC))
    {
        # Sync all paths in the backup
        $oStorageRepo->pathSync(STORAGE_REPO_BACKUP . "/${strBackupLabel}");

        foreach my $strPath ($oBackupManifest->keys(MANIFEST_SECTION_TARGET_PATH))
        {
            my $strPathSync = $oStorageRepo->pathGet(STORAGE_REPO_BACKUP . "/${strBackupLabel}/$strPath");

            # Not all paths are created for diff/incr backups, so only sync if this is a full backup or the path exists
            if ($strType eq CFGOPTVAL_BACKUP_TYPE_FULL || $oStorageRepo->pathExists($strPathSync))
            {
                $oStorageRepo->pathSync($strPathSync);
            }
        }
    }
```

Reply:
The backup label path is synced in this code.  There are also conditionals for full and exists.  Are we looking at the same code?

Yes -- these syncs used to be done later.  They have been moved up partly to be modular and partly to make sure the paths are synced for a possible resume if stop backup or archive-check fails.  Mostly just to be earlier.

Reply:
Super annoying with the wrong line numbers - I was hoping 1795 might get you close to the spot. Can you point me to the line number you're looking at that has the sync conditionals? I was in backupComplete()